src/main/java/com/aws/greengrass/easysetup/GreengrassSetup.java

```diff
@@ -358,68 +354,70 @@ void performSetup() throws IOException, DeviceConfigurationException, URISyntaxE
     private void copyTrustedPlugins(Kernel kernel, List<String> trustedPluginPaths) {
         Path trustedPluginPath;
         try {
-            trustedPluginPath = kernel.getContext().get(EZPlugins.class)
+            trustedPluginPath = kernel.getContext()
+                    .get(EZPlugins.class)
                     .withCacheDirectory(kernel.getNucleusPaths().pluginPath())
                     .getTrustedCacheDirectory();
         } catch (IOException e) {
-            logger.atError().setCause(e)
-                    .log("Caught exception while getting trusted plugins directory path");
+            logger.atError().setCause(e).log("Caught exception while getting trusted plugins directory path");
             throw new RuntimeException(e);
         }
         trustedPluginPaths.forEach(pluginPath -> {
             try {
-                Files.copy(Paths.get(pluginPath), trustedPluginPath.resolve(Utils.namePart(pluginPath)),
-                        StandardCopyOption.REPLACE_EXISTING);
+                Files.copy(Paths.get(pluginPath), trustedPluginPath
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Path traversal vulnerability detected. User-controlled input in file paths
can allow attackers to access files outside intended directories using
`../` sequences. Secure your code by using framework functions like
`getCanonicalPath()`, `normalize()`, or validating paths with
`.startsWith()` checks. Learn more:
https://cwe.mitre.org/data/definitions/22.html